packages/gestalt/src/SideNavigation.js

```diff
@@ -67,15 +72,15 @@ export default function SideNavigation({
   header,
   showBorder,
   mobileTitle,
+  collapsed,
+  onCollapse,
+  onPreview,
 }: Props): ReactNode {
-  const navigationChildren = getChildrenToArray({
-    children,
-    filterLevel: 'main',
-  });
   const { accessibilityDismissButtonLabel } = useDefaultLabelContext('SideNavigation');
   const id = useId();
   const deviceType = useDeviceType();
   const isMobile = deviceType === 'mobile';
+  const collapsible = collapsed != null;
```

Comment:
As long as `collapsed` prop is a boolean value, the `SideNavigation` is collapsible.